Shared/Samples/Show viewshed from point on map/ShowViewshedFromPointOnMapView.swift

```diff
@@ -0,0 +1,183 @@
+        await self.calculateViewshed(at: tapScreenPoint)
+    }
+    
+    /// Removes previously tapped location from overlay and draws new dot on tap location.
+    private func addGraphic(at tapPoint: Point) {
+        inputGraphicsOverlay.removeAllGraphics()
+        let graphic = Graphic(geometry: tapPoint)
+        inputGraphicsOverlay.addGraphic(graphic)
+    }
+    
+    private func calculateViewshed(at point: Point) async {
+        // Clears previously viewshed drawing.
+        self.resultGraphicsOverlay.removeAllGraphics()
+        // If there is a geoprocessing job in progress it is cancelled.
+        await self.geoprocessingJob?.cancel()
+        guard let spatialReference = point.spatialReference else { return }
+        // Creates a feature collection table based on the spatial reference for the tapped location
+        // on the map.
+        let featureCollectionTable = FeatureCollectionTable(
+            fields: [Field](),
+            geometryType: Point.self,
+            spatialReference: spatialReference
+        )
+        do {
+            // Creates a feature for the point tapped.
+            let feature = featureCollectionTable.makeFeature(geometry: point)
+            // Asynchronously adds that feature to the table.
+            try await featureCollectionTable.add(feature)
+            await self.performGeoprocessing(featureCollectionTable)
+        } catch {
+            self.error = error
+        }
+    }
+    
+    private func performGeoprocessing(_ featureCollectionTable: FeatureCollectionTable) async {
+        let params = GeoprocessingParameters(executionType: .synchronousExecute)
+        // Sets the parameters spatial reference to the point tapped on the map.
+        params.processSpatialReference = featureCollectionTable.spatialReference
+        params.outputSpatialReference = featureCollectionTable.spatialReference
+        // Create a feature with the feature collection table which has the reference to the tapped location.
+        let geoprocessingFeature = GeoprocessingFeatures(features: featureCollectionTable)
+        // Sets the observation point to the tapped location.
+        params.setInputValue(geoprocessingFeature, forKey: "Input_Observation_Point")
+        // Creates geoprocessing job and kicks off process of getting viewshed for the point.
+        geoprocessingJob = geoprocessingTask.makeJob(parameters: params)
+        geoprocessingJob?.start()
+        // Get the result of the geoprocessing job asynchronously.
+        let result = await geoprocessingJob?.result
+        switch result {
+        case .success(let output):
+            if let resultFeatures = output.outputs["Viewshed_Result"] as? GeoprocessingFeatures {
+                self.processFeatures(resultFeatures: resultFeatures)
+            }
+        case .failure(let error):
+            // Sets error to be displayed.
+            self.error = error
+        case .none:
+            // This case should never execute.
```

Comment:
None case is appearing because geoprocessingJob is now optional vs force unwrapped. The reasoning is this case should not execute and if it does execute it shouldn't crash all the samples. Let me know if this is wrong.